cartographer/sensor/range_data.cc

```diff
@@ -58,30 +58,25 @@ proto::RangeData ToProto(const RangeData& range_data) {
   proto::RangeData proto;
   *proto.mutable_origin() = transform::ToProto(range_data.origin);
   proto.mutable_returns()->Reserve(range_data.returns.size());
-  for (const Eigen::Vector3f& point : range_data.returns) {
-    *proto.add_returns() = transform::ToProto(point);
+  for (const sensor::RangefinderPoint& point : range_data.returns) {
+    *proto.add_returns() = point.ToProto();
   }
   proto.mutable_misses()->Reserve(range_data.misses.size());
-  for (const Eigen::Vector3f& point : range_data.misses) {
-    *proto.add_misses() = transform::ToProto(point);
+  for (const sensor::RangefinderPoint& point : range_data.misses) {
+    *proto.add_misses() = point.ToProto();
   }
   return proto;
 }
 
 RangeData FromProto(const proto::RangeData& proto) {
   PointCloud returns;
   returns.reserve(proto.returns().size());
-  std::transform(
-      proto.returns().begin(), proto.returns().end(),
-      std::back_inserter(returns),
-      static_cast<Eigen::Vector3f (*)(const transform::proto::Vector3f&)>(
-          transform::ToEigen));
+  std::copy(proto.returns().begin(), proto.returns().end(),
+            std::back_inserter(returns));
   PointCloud misses;
   misses.reserve(proto.misses().size());
-  std::transform(
-      proto.misses().begin(), proto.misses().end(), std::back_inserter(misses),
-      static_cast<Eigen::Vector3f (*)(const transform::proto::Vector3f&)>(
-          transform::ToEigen));
```

Comment:
I'd say that this monstrosity disappearing is the highlight of this PR

Reply:
I cannot stop being surprised that we have such code in the first place... :)